Add note about dynamically created model pickling (#5781)
pydantic/pydantic: docs/usage/models.md

      Assertion failed, must be alphanumeric [type=assertion_error, input_value='scolvi%n', input_type=str]
    """
```
!!! note

>     Pickle will work for dynamically created model only if it was defined globally

Reviewer: ```suggestion
    To pickle a dynamically created model: 
    
    - the model must be defined globally
```

>     with its class name and module where it's defined was specified.

Reviewer: I may be missing something about this. Would the following capture the same meaning?

"Pickle will work for a dynamically created model only if the model was defined globally, specifying both its class name and the module in which the class was defined."

Author: @tpdorsey I just copied the description from https://github.com/pydantic/pydantic/blob/793b484da1eb8141ce5ed0d4ec9413fb706950e7/tests/test_create_model.py#L46-L56

It may help you to understand it better.

I think the dynamic model has to have these two conditions to be pickable:

- Has to provide `__module__`
- Has to be defined globally. So, defining a dynamic model in function doesn't work

Reviewer: This works, I'll edit the above to use this language. Thx

>     with its class name and module where it's defined was specified.

Reviewer: ```suggestion
    - it must provide `__module__`
```
